src/operators/suffix.ts

```diff
@@ -127,15 +133,24 @@ export class SuffixOperator implements Operator {
    * @param obj the Object to map
    * @param maxDepth the maximum number of levels to recursive suffix child objects
    * @param currentDepth the current depth if recursively suffixing object (this is not intended to be used externally)
+   * @param totalProps the total number of props found within the object
    */
-  mapToSuffix(obj: { [key: string]: any }, currentDepth = 0): SuffixedObject {
+  mapToSuffix(obj: { [key: string]: any }, currentDepth = 0, totalProps = 0): SuffixedObject {
     const suffixedObj: SuffixedObject = {};
 
     // guard against error condition 'Cannot convert undefined or null to object'
     if (obj === undefined || obj === null) {
       return suffixedObj;
     }
 
+    // count the props to guard against super-sized object being unknowingly added to a data layer
+    // this reduces the likelihood of hitting cardinality but also prevents impacting site performance
+    const numProps = totalProps + Object.getOwnPropertyNames(obj).length;
+
+    if (numProps > this.maxProps) {
+      throw Error(`Number of object properties exceeds the limit (${this.maxProps}); increase maxProps to ${numProps}`);
```

Comment:
curious: does this mean we will abort the pipeline rather than sending partial results?

Reply:
That's a great question.  Currently, this aborts and logs and error ... we _could_ however process the 100 properties and still log.  My thinking was a missing event is much clearer that something is wrong versus partial results.  Thoughts on how you think it should work?

Reply:
The other consideration is this: a `100` event property may actually signal that what's in the data layer is anomalous.  If we do a partial recording of data, then we're ingesting "suspicious" data, which has the end result of using up cardinality.

Reply:
Ah okay yeah that makes sense, thanks for the explanation. I think this looks good then.